### ISSUE TITLE
GCSFS issues with fine res to zarr

### ISSUE THREAD

**Issue by nbren12** [VulcanClimateModeling/fv3net#1162]
I am getting many I/O errors on the read step when running the fine resolution
coarsenining with a dataflow job. The errors are all during the write step,
as opposed to reads, and have this signature:
```
Error message from worker: Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/apache_beam/utils/retry.py", line 270, in wrapper
    sleep_interval = next(retry_intervals)
StopIteration

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "apache_beam/runners/common.py", line 1239, in apache_beam.runners.common.DoFnRunner.process
  File "apache_beam/runners/common.py", line 768, in apache_beam.runners.common.PerWindowInvoker.invoke_process
  File "apache_beam/runners/common.py", line 888, in apache_beam.runners.common.PerWindowInvoker._invoke_process_per_window
  File "/Users/noah/miniconda3/envs/fv3net/lib/python3.7/site-packages/apache_beam/transforms/core.py", line 1590, in <lambda>
    wrapper = lambda x, *args, **kwargs: [fn(x, *args, **kwargs)]
  File "/usr/local/lib/python3.7/site-packages/apache_beam/utils/retry.py", line 273, in wrapper
    raise_with_traceback(exn, exn_traceback)
  File "/usr/local/lib/python3.7/site-packages/future/utils/__init__.py", line 446, in raise_with_traceback
    raise exc.with_traceback(traceback)
  File "/usr/local/lib/python3.7/site-packages/apache_beam/utils/retry.py", line 260, in wrapper
    return fun(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/budget/pipeline.py", line 17, in _write
    return mapper.write(rank)
  File "/usr/local/lib/python3.7/site-packages/xpartition.py", line 399, in write
    iOut.to_zarr(self.path, region=region)
  File "/usr/local/lib/python3.7/site-packages/xarray/core/dataset.py", line 1801, in to_zarr
    region=region,
  File "/usr/local/lib/python3.7/site-packages/xarray/backends/api.py", line 1476, in to_zarr
    writes = writer.sync(compute=compute)
  File "/usr/local/lib/python3.7/site-packages/xarray/backends/common.py", line 162, in sync
    regions=self.regions,
  File "/usr/local/lib/python3.7/site-packages/dask/array/core.py", line 945, in store
    result.compute(**kwargs)
  File "/usr/local/lib/python3.7/site-packages/dask/base.py", line 167, in compute
    (result,) = compute(self, traverse=False, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/dask/base.py", line 447, in compute
    results = schedule(dsk, keys, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/dask/local.py", line 527, in get_sync
    return get_async(apply_sync, 1, dsk, keys, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/dask/local.py", line 494, in get_async
    fire_task()
  File "/usr/local/lib/python3.7/site-packages/dask/local.py", line 466, in fire_task
    callback=queue.put,
  File "/usr/local/lib/python3.7/site-packages/dask/local.py", line 516, in apply_sync
    res = func(*args, **kwds)
  File "/usr/local/lib/python3.7/site-packages/dask/local.py", line 227, in execute_task
    result = pack_exception(e, dumps)
  File "/usr/local/lib/python3.7/site-packages/dask/local.py", line 222, in execute_task
    result = _execute_task(task, data)
  File "/usr/local/lib/python3.7/site-packages/dask/core.py", line 121, in _execute_task
    return func(*(_execute_task(a, cache) for a in args))
  File "/usr/local/lib/python3.7/site-packages/dask/array/core.py", line 3632, in store_chunk
    return load_store_chunk(x, out, index, lock, return_stored, False)
  File "/usr/local/lib/python3.7/site-packages/dask/array/core.py", line 3621, in load_store_chunk
    out[index] = np.asanyarray(x)
  File "/usr/local/lib/python3.7/site-packages/zarr/core.py", line 1115, in __setitem__
    self.set_basic_selection(selection, value, fields=fields)
  File "/usr/local/lib/python3.7/site-packages/zarr/core.py", line 1210, in set_basic_selection
    return self._set_basic_selection_nd(selection, value, fields=fields)
  File "/usr/local/lib/python3.7/site-packages/zarr/core.py", line 1501, in _set_basic_selection_nd
    self._set_selection(indexer, value, fields=fields)
  File "/usr/local/lib/python3.7/site-packages/zarr/core.py", line 1550, in _set_selection
    self._chunk_setitem(chunk_coords, chunk_selection, chunk_value, fields=fields)
  File "/usr/local/lib/python3.7/site-packages/zarr/core.py", line 1665, in _chunk_setitem
    fields=fields)
  File "/usr/local/lib/python3.7/site-packages/zarr/core.py", line 1729, in _chunk_setitem_nosync
    self.chunk_store[ckey] = cdata
  File "/usr/local/lib/python3.7/site-packages/fsspec/mapping.py", line 152, in __setitem__
    self.fs.pipe_file(key, maybe_convert(value))
  File "/usr/local/lib/python3.7/site-packages/fsspec/asyn.py", line 72, in wrapper
    return sync(self.loop, func, *args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/fsspec/asyn.py", line 53, in sync
    raise result[0]
  File "/usr/local/lib/python3.7/site-packages/fsspec/asyn.py", line 20, in _runner
    result[0] = await coro
  File "/usr/local/lib/python3.7/site-packages/gcsfs/core.py", line 1078, in _pipe_file
    self, bucket, key, data, metadata, consistency, content_type
  File "/usr/local/lib/python3.7/site-packages/gcsfs/core.py", line 1638, in simple_upload
    json_out=True,
  File "/usr/local/lib/python3.7/site-packages/gcsfs/core.py", line 556, in _call
    raise e
  File "/usr/local/lib/python3.7/site-packages/gcsfs/core.py", line 539, in _call
    self.validate_response(status, contents, path, headers)
  File "/usr/local/lib/python3.7/site-packages/gcsfs/core.py", line 1319, in validate_response
    raise HttpError(error)
gcsfs.utils.HttpError: Required, 400

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/dataflow_worker/batchworker.py", line 649, in do_work
    work_executor.execute()
  File "/usr/local/lib/python3.7/site-packages/dataflow_worker/executor.py", line 179, in execute
    op.start()
  File "dataflow_worker/shuffle_operations.py", line 63, in dataflow_worker.shuffle_operations.GroupedShuffleReadOperation.start
  File "dataflow_worker/shuffle_operations.py", line 64, in dataflow_worker.shuffle_operations.GroupedShuffleReadOperation.start
  File "dataflow_worker/shuffle_operations.py", line 79, in dataflow_worker.shuffle_operations.GroupedShuffleReadOperation.start
  File "dataflow_worker/shuffle_operations.py", line 80, in dataflow_worker.shuffle_operations.GroupedShuffleReadOperation.start
  File "dataflow_worker/shuffle_operations.py", line 84, in dataflow_worker.shuffle_operations.GroupedShuffleReadOperation.start
  File "apache_beam/runners/worker/operations.py", line 359, in apache_beam.runners.worker.operations.Operation.output
  File "apache_beam/runners/worker/operations.py", line 221, in apache_beam.runners.worker.operations.SingletonConsumerSet.receive
  File "dataflow_worker/shuffle_operations.py", line 261, in dataflow_worker.shuffle_operations.BatchGroupAlsoByWindowsOperation.process
  File "dataflow_worker/shuffle_operations.py", line 268, in dataflow_worker.shuffle_operations.BatchGroupAlsoByWindowsOperation.process
  File "apache_beam/runners/worker/operations.py", line 359, in apache_beam.runners.worker.operations.Operation.output
  File "apache_beam/runners/worker/operations.py", line 221, in apache_beam.runners.worker.operations.SingletonConsumerSet.receive
  File "apache_beam/runners/worker/operations.py", line 718, in apache_beam.runners.worker.operations.DoOperation.process
  File "apache_beam/runners/worker/operations.py", line 719, in apache_beam.runners.worker.operations.DoOperation.process
  File "apache_beam/runners/common.py", line 1241, in apache_beam.runners.common.DoFnRunner.process
  File "apache_beam/runners/common.py", line 1306, in apache_beam.runners.common.DoFnRunner._reraise_augmented
  File "apache_beam/runners/common.py", line 1239, in apache_beam.runners.common.DoFnRunner.process
  File "apache_beam/runners/common.py", line 587, in apache_beam.runners.common.SimpleInvoker.invoke_process
  File "apache_beam/runners/common.py", line 1401, in apache_beam.runners.common._OutputProcessor.process_outputs
  File "apache_beam/runners/worker/operations.py", line 221, in apache_beam.runners.worker.operations.SingletonConsumerSet.receive
  File "apache_beam/runners/worker/operations.py", line 718, in apache_beam.runners.worker.operations.DoOperation.process
  File "apache_beam/runners/worker/operations.py", line 719, in apache_beam.runners.worker.operations.DoOperation.process
  File "apache_beam/runners/common.py", line 1241, in apache_beam.runners.common.DoFnRunner.process
  File "apache_beam/runners/common.py", line 1306, in apache_beam.runners.common.DoFnRunner._reraise_augmented
  File "apache_beam/runners/common.py", line 1239, in apache_beam.runners.common.DoFnRunner.process
  File "apache_beam/runners/common.py", line 587, in apache_beam.runners.common.SimpleInvoker.invoke_process
  File "apache_beam/runners/common.py", line 1401, in apache_beam.runners.common._OutputProcessor.process_outputs
  File "apache_beam/runners/worker/operations.py", line 221, in apache_beam.runners.worker.operations.SingletonConsumerSet.receive
  File "apache_beam/runners/worker/operations.py", line 718, in apache_beam.runners.worker.operations.DoOperation.process
  File "apache_beam/runners/worker/operations.py", line 719, in apache_beam.runners.worker.operations.DoOperation.process
  File "apache_beam/runners/common.py", line 1241, in apache_beam.runners.common.DoFnRunner.process
  File "apache_beam/runners/common.py", line 1321, in apache_beam.runners.common.DoFnRunner._reraise_augmented
  File "/usr/local/lib/python3.7/site-packages/future/utils/__init__.py", line 446, in raise_with_traceback
    raise exc.with_traceback(traceback)
  File "apache_beam/runners/common.py", line 1239, in apache_beam.runners.common.DoFnRunner.process
  File "apache_beam/runners/common.py", line 768, in apache_beam.runners.common.PerWindowInvoker.invoke_process
  File "apache_beam/runners/common.py", line 888, in apache_beam.runners.common.PerWindowInvoker._invoke_process_per_window
  File "/Users/noah/miniconda3/envs/fv3net/lib/python3.7/site-packages/apache_beam/transforms/core.py", line 1590, in <lambda>
    wrapper = lambda x, *args, **kwargs: [fn(x, *args, **kwargs)]
  File "/usr/local/lib/python3.7/site-packages/apache_beam/utils/retry.py", line 273, in wrapper
    raise_with_traceback(exn, exn_traceback)
  File "/usr/local/lib/python3.7/site-packages/future/utils/__init__.py", line 446, in raise_with_traceback
    raise exc.with_traceback(traceback)
  File "/usr/local/lib/python3.7/site-packages/apache_beam/utils/retry.py", line 260, in wrapper
    return fun(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/budget/pipeline.py", line 17, in _write
    return mapper.write(rank)
  File "/usr/local/lib/python3.7/site-packages/xpartition.py", line 399, in write
    iOut.to_zarr(self.path, region=region)
  File "/usr/local/lib/python3.7/site-packages/xarray/core/dataset.py", line 1801, in to_zarr
    region=region,
  File "/usr/local/lib/python3.7/site-packages/xarray/backends/api.py", line 1476, in to_zarr
    writes = writer.sync(compute=compute)
  File "/usr/local/lib/python3.7/site-packages/xarray/backends/common.py", line 162, in sync
    regions=self.regions,
  File "/usr/local/lib/python3.7/site-packages/dask/array/core.py", line 945, in store
    result.compute(**kwargs)
  File "/usr/local/lib/python3.7/site-packages/dask/base.py", line 167, in compute
    (result,) = compute(self, traverse=False, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/dask/base.py", line 447, in compute
    results = schedule(dsk, keys, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/dask/local.py", line 527, in get_sync
    return get_async(apply_sync, 1, dsk, keys, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/dask/local.py", line 494, in get_async
    fire_task()
  File "/usr/local/lib/python3.7/site-packages/dask/local.py", line 466, in fire_task
    callback=queue.put,
  File "/usr/local/lib/python3.7/site-packages/dask/local.py", line 516, in apply_sync
    res = func(*args, **kwds)
  File "/usr/local/lib/python3.7/site-packages/dask/local.py", line 227, in execute_task
    result = pack_exception(e, dumps)
  File "/usr/local/lib/python3.7/site-packages/dask/local.py", line 222, in execute_task
    result = _execute_task(task, data)
  File "/usr/local/lib/python3.7/site-packages/dask/core.py", line 121, in _execute_task
    return func(*(_execute_task(a, cache) for a in args))
  File "/usr/local/lib/python3.7/site-packages/dask/array/core.py", line 3632, in store_chunk
    return load_store_chunk(x, out, index, lock, return_stored, False)
  File "/usr/local/lib/python3.7/site-packages/dask/array/core.py", line 3621, in load_store_chunk
    out[index] = np.asanyarray(x)
  File "/usr/local/lib/python3.7/site-packages/zarr/core.py", line 1115, in __setitem__
    self.set_basic_selection(selection, value, fields=fields)
  File "/usr/local/lib/python3.7/site-packages/zarr/core.py", line 1210, in set_basic_selection
    return self._set_basic_selection_nd(selection, value, fields=fields)
  File "/usr/local/lib/python3.7/site-packages/zarr/core.py", line 1501, in _set_basic_selection_nd
    self._set_selection(indexer, value, fields=fields)
  File "/usr/local/lib/python3.7/site-packages/zarr/core.py", line 1550, in _set_selection
    self._chunk_setitem(chunk_coords, chunk_selection, chunk_value, fields=fields)
  File "/usr/local/lib/python3.7/site-packages/zarr/core.py", line 1665, in _chunk_setitem
    fields=fields)
  File "/usr/local/lib/python3.7/site-packages/zarr/core.py", line 1729, in _chunk_setitem_nosync
    self.chunk_store[ckey] = cdata
  File "/usr/local/lib/python3.7/site-packages/fsspec/mapping.py", line 152, in __setitem__
    self.fs.pipe_file(key, maybe_convert(value))
  File "/usr/local/lib/python3.7/site-packages/fsspec/asyn.py", line 72, in wrapper
    return sync(self.loop, func, *args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/fsspec/asyn.py", line 53, in sync
    raise result[0]
  File "/usr/local/lib/python3.7/site-packages/fsspec/asyn.py", line 20, in _runner
    result[0] = await coro
  File "/usr/local/lib/python3.7/site-packages/gcsfs/core.py", line 1078, in _pipe_file
    self, bucket, key, data, metadata, consistency, content_type
  File "/usr/local/lib/python3.7/site-packages/gcsfs/core.py", line 1638, in simple_upload
    json_out=True,
  File "/usr/local/lib/python3.7/site-packages/gcsfs/core.py", line 556, in _call
    raise e
  File "/usr/local/lib/python3.7/site-packages/gcsfs/core.py", line 539, in _call
    self.validate_response(status, contents, path, headers)
  File "/usr/local/lib/python3.7/site-packages/gcsfs/core.py", line 1319, in validate_response
    raise HttpError(error)
RuntimeError: gcsfs.utils.HttpError: Required, 400 [while running 'WriteAll/Map(_write)/Map(_write)']

```

**Comment by nbren12** [comment]
So far, I am unable to reproduce this bug in a simplified setting. I opened a debugger and manually printed the data and the failing function call and collected it into a simple script:
```
import joblib
import numpy as np
import gcsfs
import asyncio


# what about this one:
loop = asyncio.get_event_loop()
self = gcsfs.GCSFileSystem(loop=loop, asynchronous=True)

async def run_test(self):
    await self.set_session()
    breakpoint()
    return await self._pipe_file(
        path="vcm-ml-scratch/private/var/folders/l1/hgxljmwd51vdlxhnnd272mqm0000gn/T/pytest-of-noah/pytest-336/test_run0/exposed_area/0.0.0.0.0",
        data=b'\x02\x01!\x04p,\x00\x00p,\x00\x00\x15\x06\x00\x00\x14\x00\x00\x00}\x01\x00\x00\xff\x01\xf8\xb4\xd9JTQ\xb8\x96\xc6K\xa0\x00\x00\x00\x00\x00\x05\x00\x01\xbf\xe6p\x93\x80\x99\xb0\xbe\xf8\x1aH\x08\x1f\x00\x01\x01\x02\x00\xb2{\xa8\xf4\x16bc\x16p\xf0\x1cV\x11\x00\x0f\x02\x00\x00\xbfL:\x8c\xa4#4\x860L\x86\x95\x1f\x00\x01\x01\x02\x00\xb2\x10\x85@\xdf\xdd\xfd:\xf0\xc7R\xbc\x11\x00\x0f\x02\x00\x00\xbf\x13r8\xda\xcc\x1c=\xdb\x15\xbf\xaa\x1f\x00\x01\x01\x02\x00\xb2\x8cZ\x08\xdd^\xc7\x85Xv\xc8\xf8\x11\x00\x0f\x02\x00\x00\xbf\xda\x80\xaf\xcb\xb1\x96\x00\xc4l\xccN\x1f\x00\x01\x01\x02\x00\xb2\xac\xe4\xedhx\xb4\x82d\x008:\x11\x00\x0f\x02\x00\x00\xbf\xb1\x8bl\xa8\\\x8dH\x87\x05\xec \x1f\x00\x01\x01\x02\x00\xb28\nz\xe7}\xc0\xd2B .f\x11\x00\x0f\x02\x00\x00\xbf~6\xf4\x04\xb9\x98S\xb8\x11\x87\x14\x1f\x00\x01\x01\x02\x00\xb2\xc6@\x98\xeepO\xed\xe2\x84\x8a0\x11\x00\x0f\x02\x00\x00\xbf\x18\nd\xe8\xd8\xa27/40l\x1f\x00\x01\x01\x02\x00\xb2\xfc\xb8\xdc>$\x95o\x02"\x9a\xcd\x11\x00\x0f\x02\x00\x00\xbf]\x89\x12\xae\xf3\x16]d\x0c\xee\x91\x1f\x00\x01\x01\x02\x00\xb2:\xba^|\xba\xf8\x07%Y\x94\x08\x11\x00\x0f\x02\x00\x00\xbf|[\xe0{\x89D \xc9\x03\xf8\x98\x1f\x00\x01\x01\x02\x00\xb2<Br\x9e\xceo\xff\xb0\xa0\xc8\x90\x11\x00\x0f\x02\x00\x00\xbf\xdd\xaa\x8d\xf3\x82I\x0ckw\x930\x1f\x00\x01\x01\x02\x00\x0f\xd0\x02\xff\xff\xff\xff\xff\xff\xff\xff<P\x00\x00\x00\x00\x00}\x01\x00\x00\xff\x01\xe5\x8f\xa8V|\xe2.\xbb\xb9\x17\x1e\x00\x00\x00\x00\x00\x05\x00\x01\xbf\xa1\x83/\x12B\xb4\xd8E\xa7\xb2R\x1f\x00\x01\x01\x02\x00\xb2\x96F[\x97\x05\xf9\xde>\xf6\x9a.\x11\x00\x0f\x02\x00\x00\xbf\x189\xaa)^\x07\xb1\xdd*o:\x1f\x00\x01\x01\x02\x00\xb2=\xfb\xaeVu\xcc\x12\x80\xad\x86\xcc\x11\x00\x0f\x02\x00\x00\xbf7\x8cp\xb8\xa96\xd4_\xadCO\x1f\x00\x01\x01\x02\x00\xb2\xfb\xb7n\x7f~\xb7\xa3\xee\xd0\xd0y\x11\x00\x0f\x02\x00\x00\xbfC?\xc2\xe9O@\xbc\x12\x1b\xdd*\x1f\x00\x01\x01\x02\x00\xb2\x833\xfeBG\x1a\xb7XLy\xd7\x11\x00\x0f\x02\x00\x00\xbf\xaa\xa5a\xbeL (\xe5Q\xado\x1f\x00\x01\x01\x02\x00\xb2u\x82\x89\xc7;\x19\'.\xe3s\xfb\x11\x00\x0f\x02\x00\x00\xbf\x1aFe\xb5\x171\xffS\xeb\xe7h\x1f\x00\x01\x01\x02\x00\xb2\xa1.g\x03\xe3\xec\xbdGN[\x1d\x11\x00\x0f\x02\x00\x00\xbf\x08\xeay\x0b\x90\xee\xfa\x9a\x80\xbd\xfb\x1f\x00\x01\x01\x02\x00\xb2\x9a\xe4\x819\x14\xc4\xa7\x95\xc0\xf6R\x11\x00\x0f\x02\x00\x00\xbf\xf2#\xb1\xeb\x92\x8c\xa3\xc8M\xfd\xd6\x1f\x00\x01\x01\x02\x00\xb2\xd6\xadw\xf2\x08s\x0cq\xbb\xbb{\x11\x00\x0f\x02\x00\x00\xbf\x02\xf0\xff\x87\x88\xabq@\xb1\xf3\x0e\x1f\x00\x01\x01\x02\x00\xb2\x19b8\x87\xee\x97\xa7qyO\xaa\x11\x00\x0f\x02\x00\x00\xbfa=[c\x96\x02\x89\xfa\xcet-\x1f\x00\x01\x01\x02\x00\x0f\xd0\x02\xff\xff\xff\xff\xff\xff\xff\xff<P\x00\x00\x00\x00\x00}\x01\x00\x00\xff\x01?\xb4m$\xbf\x9f\xc2\x8e\xfa\xde\x1e\xc0\xc0\xc0\xc0\xc0\x05\x00\x01\xbf\x92(\x01\r\x90\x97\xe0"\x80\xa9\xf5\x1f\x00\x01\x01\x02\x00\xb2\xdf\xaf\xf7\x1ek\x1b\xd9\x0c\xc5\xad\x81\x11\x00\x0f\x02\x00\x00\xbf\xb91\x92V\x08\xee\xd4j\x88\x86\x88\x1f\x00\x01\x01\x02\x00\xb2K\xd3\x0b\x87G-\x16"\x81\x100\x11\x00\x0f\x02\x00\x00\xbf\xf9M\xb5\xa0X\xfd\x14h\xcei \x1f\x00\x01\x01\x02\x00\xb2\x97H\xf0h\x16\xd9\xb7g\x14\x811\x11\x00\x0f\x02\x00\x00\xbf\x8c\xb7\x1d\xfd\x1b7\x04{\x05l \x1f\x00\x01\x01\x02\x00\xb2\xa4\x7f\xb1\x0e\x7f\x99\xed\rxb\xe8\x11\x00\x0f\x02\x00\x00\xbf\xda.\xf4\xe9\xd0\xdd\x002\x98\xd8\xb2\x1f\x00\x01\x01\x02\x00\xb2\x92\xc7\xe9\xb1r\x036\x1d\xa8\xbd\xd0\x11\x00\x0f\x02\x00\x00\xbf\x14l\x01k\xcd\xa3\x15\n\x8f\x1f\xe1\x1f\x00\x01\x01\x02\x00\xb2\xe5\xa2\x80\xbb\xd0B\xffR\x02-$\x11\x00\x0f\x02\x00\x00\xbfQ)\xde\xa7w\xb5\xb1\x88\xceii\x1f\x00\x01\x01\x02\x00\xb2|\xa3H9\xaa\xb4l1\x84\n|\x11\x00\x0f\x02\x00\x00\xbf\xecQkT\xaa\xaeL\xaa\xb1\xf4\xc8\x1f\x00\x01\x01\x02\x00\xb2W\xe3\xbd\x9f\xb1\xaft2\x91V\xa6\x11\x00\x0f\x02\x00\x00\xbf\x8b\xcd`\xb7\x9c\x9c\x1f$>\x0b\x1b\x1f\x00\x01\x01\x02\x00\xb2!\x816)C\xee\x18\x0f\xcd\xc0\x9e\x11\x00\x0f\x02\x00\x00\xbf\x12{\x0c\xcc\xaa\x96\x91\r\x92Th\x1f\x00\x01\x01\x02\x00\x0f\xd0\x02\xff\xff\xff\xff\xff\xff\xff\xff<P\xc0\xc0\xc0\xc0\xc0z\x01\x00\x00\xff\x01\xc5\xc4E\xc6\xc5\xc4\xc3EEED\x7f\x7f\x7f\x7f\x7f\x05\x00\x01\xbfE\xc5\xc5\xc2ED\xc5D\xc5\xc5\xc4\x1f\x00\x01\x01\x02\x00\xb2\xc5DD\xc5E\xc5E\xc5\xc2EE\x11\x00\x0f\x02\x00\x00\xbfD\xc5\xc4\xc5\xc5D\xc4\xc4\xc5\xc4E\x1f\x00\x01\x01\x02\x00\xb2E\xc5\xc3D\xc6E\xc6CEEF\x11\x00\x0f\x02\x00\x00\xbf\xc4\xc5C\xc5\xc4\xc4EDDE\xc6\x1f\x00\x01\x01\x02\x00\xb2\xc5\xc5\xc2\xc5F\xc5\xc4\xc4FE\xc5\x11\x00\x0f\x02\x00\x00\xbf\xc3\xc4F\xc5\xc5\xc5\xc5E\xc6\xc3F\x1f\x00\x01\x01\x02\x00\xb2DDEEEE\xc4\xc6\xc4\xc4D\x11\x00\x0f\x02\x00\x00\xbf\xc5\xc4DCE\xc4E\xc5E\xc4E\x1f\x00\x01\x01\x02\x00\xb2\xc5\xc3\xc5D\xc5DE\xc5DDE\x11\x00\x0f\x02\x00\x00\xbfD\xc5F\xc5ED\xc5EE\xc5\xc5\x1f\x00\x01\x01\x02\x00\xb2\xc4E\xc4\xc5BEE\xc6\xc5\xc5E\x11\x00\x0f\x02\x00\x00\x00\xb4\x00\x7f\xc4\xc5\xc5E\xc5CD\x1e\x00\x00\x02\x02\x00p\xc4E\xc6DE\xc4E\xdd\x00\x02\x11\x00\x0f\x02\x00\x00\xbfE\xc5\xc5EE\xc5\xc5E\xc4\xc4D\x1f\x00\x01\x01\x02\x00\xb2\xc5\xc5DE\xc5\xc5\xc6\xc4\xc5D\xc4\x11\x00\x0f\x02\x00\x00\xbfEDD\xc5\xc4EE\xc6EE\xc3\x1f\x00\x01\x01\x02\x00\xb2F\xc5\xc6EE\xc5\xc5\xc3\xc5\xc4E\x11\x00\x0f\x02\x00\x00\x00\x90\x00\x7f\xc5\xc5\xc4EEEE\x1e\x00\x00\x02\x02\x00\x0f\xd0\x02\xff\xff\xff\xff\xff\xff\xff\xff<P\x7f\x7f\x7f\x7f\x7f',
        metadata=None,
        consistency="none",
        content_type="application/octet-stream",
        chunksize=52428800,
    )

arg = run_test(self)
loop.run_until_complete(arg)
```
This script succeeds without a problem. This suggests the issue is related to running in the context of Apache beam.

**Comment by nbren12** [comment]
It works in an apache beam pipeline too:
```
import joblib
import numpy as np
import gcsfs
import asyncio
import apache_beam
from fv3net.pipelines.common import FunctionSource 

# this level works

def write(args):
    out, index, x = args
    out[index] = np.asanyarray(x)
    print(out[index])


with apache_beam.Pipeline() as p:
    (
        p
        | FunctionSource( joblib.load,
            "/Users/noah/04-27-test-data.pkl",
        )
        | apache_beam.Map(write)
    )
```

**Comment by nbren12** [comment]
This problem seems isolated to the case where there is a medium sized dask graph in the input. Perhaps the gcsfs filesystem object is opening too many event loops?

**Comment by nbren12** [comment]
Closed by #1172
